internal/mesh/internal/types/proxy_state_template.go

```diff
@@ -92,6 +91,11 @@ func ValidateProxyStateTemplate(res *pbresource.Resource) error {
 				}))
 			}
 
+			//TODO: Call exports.go proxy of validators.go's protocol validation code, ensure it's set to not-unspecified enum value
+			// (That export also needs to be added, see examples there)
+			//TODO: Add golden test coverage of the changes made to Cluster to include protocol
+			//TODO: Clean up commit message
```

Comment:
@jmurret added TODOs here. The linter errors are just from the need to propagate this change through to the converter